### PR TITLE
op-node: Log on L1 Info deposit mismatch

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Log significantly more information if the invalid transaction is the L1 info transaction.
This is useful for debugging.

### CODE REVIEW

File: op-node/rollup/derive/engine_consolidate.go [hunk 1]
```diff
@@ -26,6 +28,27 @@ func AttributesMatchBlock(attrs *eth.PayloadAttributes, parentHash common.Hash,
 	}
 	for i, otx := range attrs.Transactions {
 		if expect := block.Transactions[i]; !bytes.Equal(otx, expect) {
+			if i == 0 {
```
Comment (tynes) [inline op-node/rollup/derive/engine_consolidate.go:31]: This is a pretty unwieldy piece of code, what do you think about putting it into a function so its not all in the hot code path?

File: op-node/rollup/derive/engine_consolidate.go [hunk 2]
```diff
@@ -26,6 +28,27 @@ func AttributesMatchBlock(attrs *eth.PayloadAttributes, parentHash common.Hash,
...
+							"safe_l1_number", safeInfo.Number, "safe_l1_hash", safeInfo.BlockHash,
+							"safe_l1_time", safeInfo.Time, "safe_seq_num", safeInfo.SequenceNumber,
+							"unsafe_l1_number", unsafeInfo.Number, "unsafe_l1_hash", unsafeInfo.BlockHash,
+							"unsafe_l1_time", unsafeInfo.Time, "unsafe_seq_num", unsafeInfo.SequenceNumber)
+					} else {
+						l.Warn("failed to umarshal l1 info", "errSafe", errSafe, "errUnsafe", errUnsafe)
+					}
+				} else {
+					l.Warn("failed to umarshal tx", "errSafe", errSafe, "errUnsafe", errUnsafe)
+				}
```
Comment (ajsutton) [inline op-node/rollup/derive/engine_consolidate.go:50]: nit: Would it be worth extracting this as a separate method? Then the main business logic would be easier to read, but also you could probably avoid the deep nesting here by returning early on errors.

But by all means ignore me, my experience with styling go code involves watching about three YouTube videos. :)

Comment (trianglesphere) [reply op-node/rollup/derive/engine_consolidate.go:50]: good structure is good structure regardless of language. I can pull this out into it's own method - that'll help a bunch. We really shouldn't hit this error case, but I've had to recreate this log enough that I finally wanted to merge it.

File: op-node/rollup/derive/engine_consolidate.go (hunk 1 above)
Comment (trianglesphere) [reply op-node/rollup/derive/engine_consolidate.go:32]: done

File: op-node/rollup/derive/engine_consolidate.go [hunk 3]
```diff
@@ -37,3 +43,33 @@ func AttributesMatchBlock(attrs *eth.PayloadAttributes, parentHash common.Hash,
...
+
+// logL1InfoTxns reports the values from the L1 info tx when they differ to aid
+// debugging. This check is the one that has been most frequently triggered.
+func logL1InfoTxns(l log.Logger, l2Number, l2Timestamp uint64, safeTx, unsafeTx hexutil.Bytes) {
+	// First decode into *types.Transaction to get the tx data.
+	var safeTxValue, unsafeTxValue types.Transaction
+	errSafe := (&safeTxValue).UnmarshalBinary(safeTx)
+	errUnsafe := (&unsafeTxValue).UnmarshalBinary(unsafeTx)
+	if errSafe != nil || errUnsafe != nil {
+		l.Error("failed to umarshal tx", "errSafe", errSafe, "errUnsafe", errUnsafe)
```
Comment (protolambda) [inline op-node/rollup/derive/engine_consolidate.go:55]: Needs a return if the transactions failed to decode.

File: op-node/rollup/derive/engine_consolidate.go [hunk 4]
```diff
@@ -37,3 +43,33 @@ func AttributesMatchBlock(attrs *eth.PayloadAttributes, parentHash common.Hash,
...
+	}
+
+	// Then decode the ABI encoded parameters
+	var safeInfo, unsafeInfo L1BlockInfo
+	errSafe = (&safeInfo).UnmarshalBinary(safeTxValue.Data())
+	errUnsafe = (&unsafeInfo).UnmarshalBinary(unsafeTxValue.Data())
+	if errSafe != nil || errUnsafe != nil {
+		l.Error("failed to umarshal l1 info", "errSafe", errSafe, "errUnsafe", errUnsafe)
```
Comment (protolambda) [inline op-node/rollup/derive/engine_consolidate.go:63]: ditto

File: op-node/rollup/derive/engine_consolidate.go (hunk 3 above)
Comment (trianglesphere) [reply op-node/rollup/derive/engine_consolidate.go:55]: nice catch

**Review by protolambda — APPROVED** [review]

**Pull request merged**